src/ripple/app/consensus/LedgerConsensus.cpp

```diff
@@ -372,7 +374,8 @@ class LedgerConsensusImp
       @param map      the transaction set.
       @param acquired true if we have acquired the transaction set.
     */
-    void mapComplete (uint256 const& hash, SHAMap::ref map, bool acquired)
+    void mapComplete (uint256 const& hash, std::shared_ptr<SHAMap> const& map,
+                      bool acquired)
```

Comment:
Having to learn one thing once is *way* better than three constant pains this change will cause. First, the increased verbosity every time it's used. Second, the inability to easily find the places where shared pointers are passed by reference. Third, the risk that the meaning will be missed -- a const reference to a shared pointer creates a particular unified semantic.